core/unit_test/incremental/Test17_CompleteAtomic.hpp

```diff
@@ -0,0 +1,127 @@
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <Kokkos_Core.hpp>
+#include <random>
+#include <gtest/gtest.h>
+
+/// @Kokkos_Feature_Level_Required:17
+// Incremental test for atomic views.
+// In this test we sort N integers into num_buckets number of buckets based on
+// their rermainder, i.e., a histogram based on remainder. Since the number of
+// integers is greater than the number of buckets, we use atomic views for the
+// sorted histogram.
+
+namespace Test {
+
+using value_type = int;
+const int N      = 50;
+#define num_buckets 10
+
+template <class ExecSpace>
+struct TestAtomicView {
+  // 1D  View of int
+  using View = typename Kokkos::View<value_type *, ExecSpace>;
+
+  // 1D atomic view
+  using atomic_view =
+      typename Kokkos::View<value_type *, ExecSpace,
+                            Kokkos::MemoryTraits<Kokkos::Atomic> >;
+
+  void atomicView() {
+    // Use default_random_engine object to introduce randomness.
+    std::default_random_engine generator;
+    // Initialize uniform_int_distribution class.
+    std::uniform_int_distribution<int> distribution(0, N);
+
+    // Device and Host views of N number of integers
+    View d_data("deviceData_1D", N);
+    auto h_data = create_mirror_view(d_data);
+
+    // Atomic Device and Host views of histogram
+    atomic_view d_hist("histogram", num_buckets);
+    auto h_hist = create_mirror_view(d_hist);
+
+    // An array to store correct results for verification
+    int *correct_results = new int[num_buckets];
+
+    // Initialize host side histogram arrays
+    for (int i = 0; i < num_buckets; ++i) {
+      h_hist(i)          = 0;
+      correct_results[i] = 0;
+    }
+
+    // Fill host data with integers from the distribution object.
+    for (int i = 0; i < N; ++i) h_data(i) = distribution(generator);
+
+    // Copy data from host to device
+    Kokkos::deep_copy(d_data, h_data);
+    Kokkos::deep_copy(d_hist, h_hist);
+
+    // Update histogram
+    Kokkos::parallel_for(
+        N, KOKKOS_LAMBDA(const int i) { d_hist(d_data(i) % num_buckets)++; });
```

Comment:
This looks suspicious. You don't specify the execution space so it is running on the host but your accessing data on the device.